In [1]:
import pandas as pd

In [3]:
import matplotlib.pyplot as plt

In [5]:
games=pd.read_csv(r'C:\Users\hp\Data Analyst\csv\Online_game.csv')

In [7]:
games

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

### Description

* This dataset appears to represent video game sales data, containing information about 16,598 video games. 
* Each row corresponds to a specific video game, and the dataset includes the following columns
   * Rank: A unique integer identifier that ranks the video games based on a specific criterion (likely sales    * Name: The name of the video ga
    * Platform: The gaming platform on which the video game was released (e.g., PS4, Xbox One, 
    * Year: The ye game was released. This column has some missing values, with 16,327 non-null entries out of 16,598.
    * Genre: The genre of the video game (e.g., Action, Sports, RPG).
    * Publisher: The company that published the video game. There are a few missing values, with 16,540 non-null entries.
    * NA_Sales: Sales in North America, measured in millions of units.
    * EU_Sales: Sales in Europe, measured in millions of units.
    * JP_Sales: Sales in Japan, measured in millions of units.
    *
Other_Sales: Sales in regions other than North America, Europe, and Japan, measured in millions of units.
    * Global_Sales: The total global sales of the video game, measured in millions of units.ar the

In [17]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


In [20]:
games.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [26]:
games.fillna(0,inplace=True)

In [28]:
games.isnull().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

In [32]:
games.duplicated().sum()

0

In [34]:
games.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16598.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   1973.647307      0.264667      0.146652      0.077782   
std     4791.853933    254.346809      0.816683      0.505351      0.309291   
min        1.000000      0.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

In [36]:
games.describe().T

count         mean          std   min      25%      50%  \
Rank          16598.0  8300.605254  4791.853933  1.00  4151.25  8300.50   
Year          16598.0  1973.647307   254.346809  0.00  2003.00  2007.00   
NA_Sales      16598.0     0.264667     0.816683  0.00     0.00     0.08   
EU_Sales      16598.0     0.146652     0.505351  0.00     0.00     0.02   
JP_Sales      16598.0     0.077782     0.309291  0.00     0.00     0.00   
Other_Sales   16598.0     0.048063     0.188588  0.00     0.00     0.01   
Global_Sales  16598.0     0.537441     1.555028  0.01     0.06     0.17   

                   75%       max  
Rank          12449.75  16600.00  
Year           2010.00   2020.00  
NA_Sales          0.24     41.49  
EU_Sales          0.11     29.02  
JP_Sales          0.04     10.22  
Other_Sales       0.04     10.57  
Global_Sales      0.47     82.74

In [38]:
games

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [46]:
games['Publisher'].unique()

array(['Nintendo', 'Microsoft Game Studios', 'Take-Two Interactive',
       'Sony Computer Entertainment', 'Activision', 'Ubisoft',
       'Bethesda Softworks', 'Electronic Arts', 'Sega', 'SquareSoft',
       'Atari', '505 Games', 'Capcom', 'GT Interactive',
       'Konami Digital Entertainment',
       'Sony Computer Entertainment Europe', 'Square Enix', 'LucasArts',
       'Virgin Interactive', 'Warner Bros. Interactive Entertainment',
       'Universal Interactive', 'Eidos Interactive', 'RedOctane',
       'Vivendi Games', 'Enix Corporation', 'Namco Bandai Games',
       'Palcom', 'Hasbro Interactive', 'THQ', 'Fox Interactive',
       'Acclaim Entertainment', 'MTV Games', 'Disney Interactive Studios',
       0, 'Majesco Entertainment', 'Codemasters', 'Red Orb', 'Level 5',
       'Arena Entertainment', 'Midway Games', 'JVC', 'Deep Silver',
       '989 Studios', 'NCSoft', 'UEP Systems', 'Parker Bros.', 'Maxis',
       'Imagic', 'Tecmo Koei', 'Valve Software', 'ASCII Entertainment',
  

## -Q1Retrieve the details of nintendo publishers

In [60]:
nint=games[games['Publisher']=='Nintendo']
nint

Rank                                               Name Platform  \
0          1                                         Wii Sports      Wii   
1          2                                  Super Mario Bros.      NES   
2          3                                     Mario Kart Wii      Wii   
3          4                                  Wii Sports Resort      Wii   
4          5                           Pokemon Red/Pokemon Blue       GB   
...      ...                                                ...      ...   
16269  16272                           Slide Adventure: Mag Kid       DS   
16357  16360               Mario vs. Donkey Kong: Tipping Stars     WiiU   
16456  16459                           Art Academy: Home Studio     WiiU   
16473  16476                                    Captain Rainbow      Wii   
16542  16545  Mario & Luigi: Paper Jam & Mario Kart 7 Double...      3DS   

         Year         Genre Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports  Nintendo     41.49     29.02      3.77   
1      1985.0      Platform  Nintendo     29.08      3.58      6.81   
2      2008.0        Racing  Nintendo     15.85     12.88      3.79   
3      2009.0        Sports  Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing  Nintendo     11.27      8.89     10.22   
...       ...           ...       ...       ...       ...       ...   
16269  2007.0        Action  Nintendo      0.00      0.00      0.01   
16357  2015.0        Puzzle  Nintendo      0.00      0.00      0.01   
16456  2015.0          Misc  Nintendo      0.00      0.00      0.01   
16473  2008.0     Adventure  Nintendo      0.00      0.00      0.01   
16542  2015.0          Misc  Nintendo      0.00      0.00      0.01   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16269         0.00          0.01  
16357         0.00          0.01  
16456         0.00          0.01  
16473         0.00          0.01  
16542         0.00          0.01  

[703 rows x 11 columns]

In [86]:
games[games['Publisher'(games['Global_Sales'].max())

SyntaxError: incomplete input (1727968879.py, line 1)

## -Q2retrieve the games that is developed by nintendo

In [62]:
nint['Name'].unique()

array(['Wii Sports', 'Super Mario Bros.', 'Mario Kart Wii',
       'Wii Sports Resort', 'Pokemon Red/Pokemon Blue', 'Tetris',
       'New Super Mario Bros.', 'Wii Play', 'New Super Mario Bros. Wii',
       'Duck Hunt', 'Nintendogs', 'Mario Kart DS',
       'Pokemon Gold/Pokemon Silver', 'Wii Fit', 'Wii Fit Plus',
       'Super Mario World',
       'Brain Age: Train Your Brain in Minutes a Day',
       'Pokemon Diamond/Pokemon Pearl', 'Super Mario Land',
       'Super Mario Bros. 3', 'Pokemon Ruby/Pokemon Sapphire',
       'Pokemon Black/Pokemon White',
       'Brain Age 2: More Training in Minutes a Day',
       'Pokémon Yellow: Special Pikachu Edition', 'Pokemon X/Pokemon Y',
       'Super Smash Bros. Brawl', 'Animal Crossing: Wild World',
       'Mario Kart 7', 'Pokemon HeartGold/Pokemon SoulSilver',
       'Super Mario 64', 'Super Mario Galaxy',
       'Pokemon Omega Ruby/Pokemon Alpha Sapphire',
       'Super Mario Land 2: 6 Golden Coins', 'Super Mario 3D Land',
       'Super Mario

In [68]:
nint['Name']

0                                               Wii Sports
1                                        Super Mario Bros.
2                                           Mario Kart Wii
3                                        Wii Sports Resort
4                                 Pokemon Red/Pokemon Blue
                               ...                        
16269                             Slide Adventure: Mag Kid
16357                 Mario vs. Donkey Kong: Tipping Stars
16456                             Art Academy: Home Studio
16473                                      Captain Rainbow
16542    Mario & Luigi: Paper Jam & Mario Kart 7 Double...
Name: Name, Length: 703, dtype: object

## -Q3publisher with highest sale in  global

In [116]:
games.groupby(['Publisher'])['Global_Sales'].max().idxmax()

'Nintendo'

In [140]:
games.groupby(['Publisher'])['Global_Sales'].max().sort_values(ascending=False).idxmax()

'Nintendo'

## -Q4In which region the publisher nintendo has high sales

In [127]:
games

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [145]:
high=games.groupby(['Publisher'])

In [149]:
high=high.get_group('Nintendo')

C:\Users\hp\AppData\Local\Temp\ipykernel_716\2118574242.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  high=high.get_group('Nintendo')


In [151]:
high

Rank                                               Name Platform  \
0          1                                         Wii Sports      Wii   
1          2                                  Super Mario Bros.      NES   
2          3                                     Mario Kart Wii      Wii   
3          4                                  Wii Sports Resort      Wii   
4          5                           Pokemon Red/Pokemon Blue       GB   
...      ...                                                ...      ...   
16269  16272                           Slide Adventure: Mag Kid       DS   
16357  16360               Mario vs. Donkey Kong: Tipping Stars     WiiU   
16456  16459                           Art Academy: Home Studio     WiiU   
16473  16476                                    Captain Rainbow      Wii   
16542  16545  Mario & Luigi: Paper Jam & Mario Kart 7 Double...      3DS   

         Year         Genre Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports  Nintendo     41.49     29.02      3.77   
1      1985.0      Platform  Nintendo     29.08      3.58      6.81   
2      2008.0        Racing  Nintendo     15.85     12.88      3.79   
3      2009.0        Sports  Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing  Nintendo     11.27      8.89     10.22   
...       ...           ...       ...       ...       ...       ...   
16269  2007.0        Action  Nintendo      0.00      0.00      0.01   
16357  2015.0        Puzzle  Nintendo      0.00      0.00      0.01   
16456  2015.0          Misc  Nintendo      0.00      0.00      0.01   
16473  2008.0     Adventure  Nintendo      0.00      0.00      0.01   
16542  2015.0          Misc  Nintendo      0.00      0.00      0.01   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16269         0.00          0.01  
16357         0.00          0.01  
16456         0.00          0.01  
16473         0.00          0.01  
16542         0.00          0.01  

[703 rows x 11 columns]

In [159]:
high[['NA_Sales','EU_Sales','JP_Sales','Other_Sales']].max()

NA_Sales       41.49
EU_Sales       29.02
JP_Sales       10.22
Other_Sales     8.46
dtype: float64

In [157]:
high[['NA_Sales','EU_Sales','JP_Sales','Other_Sales']].max().idxmax()

'NA_Sales'

## -Q5Retrieve the game that has highest sale in 2010

In [187]:
games['Year']=games['Year'].astype('int64')

In [198]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16598 non-null  int64  
 4   Genre         16598 non-null  object 
 5   Publisher     16598 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(5), int64(2), object(4)
memory usage: 1.4+ MB


In [214]:
highgame=games.groupby(['Year'])

In [216]:
highgame

In [220]:
highgame=highgame.get_group(2010)
highgame

C:\Users\hp\AppData\Local\Temp\ipykernel_716\2972908449.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  highgame=highgame.get_group(2010)


Rank                                               Name Platform  \
15        16                                 Kinect Adventures!     X360   
26        27                        Pokemon Black/Pokemon White       DS   
31        32                            Call of Duty: Black Ops     X360   
40        41                            Call of Duty: Black Ops      PS3   
54        55                                     Gran Turismo 5      PS3   
...      ...                                                ...      ...   
16437  16440              The Cages: Pro Style Batting Practice      Wii   
16478  16481                 Ishin no Arashi: Shippuu Ryuumeden       DS   
16487  16490          Slotter Mania P: Tetsuya Shinjuku vs Ueno      PSP   
16501  16504  Transformers: War for Cybertron (XBox 360, PS3...       PC   
16596  16599                                         Know How 2       DS   

       Year         Genre                     Publisher  NA_Sales  EU_Sales  \
15     2010          Misc        Microsoft Game Studios     14.97      4.94   
26     2010  Role-Playing                      Nintendo      5.57      3.28   
31     2010       Shooter                    Activision      9.67      3.73   
40     2010       Shooter                    Activision      5.98      4.44   
54     2010        Racing   Sony Computer Entertainment      2.96      4.88   
...     ...           ...                           ...       ...       ...   
16437  2010        Sports  Konami Digital Entertainment      0.01      0.00   
16478  2010      Strategy                    Tecmo Koei      0.00      0.00   
16487  2010          Misc                        Dorart      0.00      0.00   
16501  2010       Shooter                    Activision      0.01      0.00   
16596  2010        Puzzle                      7G//AMES      0.00      0.01   

       JP_Sales  Other_Sales  Global_Sales  
15         0.24         1.67         21.82  
26         5.65         0.82         15.32  
31         0.11         1.13         14.64  
40         0.48         1.83         12.73  
54         0.81         2.12         10.77  
...         ...          ...           ...  
16437      0.00         0.00          0.01  
16478      0.01         0.00          0.01  
16487      0.01         0.00          0.01  
16501      0.00         0.00          0.01  
16596      0.00         0.00          0.01  

[1259 rows x 11 columns]

In [224]:
highgame=highgame.groupby(['Name'])

In [234]:
highgame[['NA_Sales','EU_Sales','JP_Sales','Other_Sales']].max().idxmax()

NA_Sales                Kinect Adventures!
EU_Sales                Kinect Adventures!
JP_Sales       Pokemon Black/Pokemon White
Other_Sales                 Gran Turismo 5
dtype: object

In [236]:
highgame['Global_Sales'].max().idxmax()

'Kinect Adventures!'

## Q6Retrive the games name and publisher with genre is shooter

In [255]:
genre=games.groupby(['Genre'])

In [257]:
genre=genre.get_group('Shooter')
genre

C:\Users\hp\AppData\Local\Temp\ipykernel_716\220463387.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  genre=genre.get_group('Shooter')


Rank                                               Name Platform  \
9         10                                          Duck Hunt      NES   
29        30                     Call of Duty: Modern Warfare 3     X360   
31        32                            Call of Duty: Black Ops     X360   
33        34                          Call of Duty: Black Ops 3      PS4   
34        35                         Call of Duty: Black Ops II      PS3   
...      ...                                                ...      ...   
16501  16504  Transformers: War for Cybertron (XBox 360, PS3...       PC   
16569  16572                                 Resident Evil 4 HD     XOne   
16577  16580                                          Damnation       PC   
16585  16588                                             Breach       PC   
16594  16597                      Men in Black II: Alien Escape       GC   

       Year    Genre    Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
9      1984  Shooter     Nintendo     26.93      0.63      0.28         0.47   
29     2011  Shooter   Activision      9.03      4.28      0.13         1.32   
31     2010  Shooter   Activision      9.67      3.73      0.11         1.13   
33     2015  Shooter   Activision      5.77      5.81      0.35         2.31   
34     2012  Shooter   Activision      4.99      5.88      0.65         2.52   
...     ...      ...          ...       ...       ...       ...          ...   
16501  2010  Shooter   Activision      0.01      0.00      0.00         0.00   
16569  2016  Shooter       Capcom      0.01      0.00      0.00         0.00   
16577  2009  Shooter  Codemasters      0.00      0.01      0.00         0.00   
16585  2011  Shooter    Destineer      0.01      0.00      0.00         0.00   
16594  2003  Shooter   Infogrames      0.01      0.00      0.00         0.00   

       Global_Sales  
9             28.31  
29            14.76  
31            14.64  
33            14.24  
34            14.03  
...             ...  
16501          0.01  
16569          0.01  
16577          0.01  
16585          0.01  
16594          0.01  

[1310 rows x 11 columns]

In [261]:
genre[['Name','Publisher','Genre']]

Name    Publisher    Genre
9                                              Duck Hunt     Nintendo  Shooter
29                        Call of Duty: Modern Warfare 3   Activision  Shooter
31                               Call of Duty: Black Ops   Activision  Shooter
33                             Call of Duty: Black Ops 3   Activision  Shooter
34                            Call of Duty: Black Ops II   Activision  Shooter
...                                                  ...          ...      ...
16501  Transformers: War for Cybertron (XBox 360, PS3...   Activision  Shooter
16569                                 Resident Evil 4 HD       Capcom  Shooter
16577                                          Damnation  Codemasters  Shooter
16585                                             Breach    Destineer  Shooter
16594                      Men in Black II: Alien Escape   Infogrames  Shooter

[1310 rows x 3 columns]

## Q7List out the racing game that is published by Sony Computer Entertainment

In [280]:
race=games.groupby(['Genre'])

In [282]:
race=race.get_group('Racing')
race

C:\Users\hp\AppData\Local\Temp\ipykernel_716\3604274133.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  race=race.get_group('Racing')


Rank                                               Name Platform  \
2          3                                     Mario Kart Wii      Wii   
11        12                                      Mario Kart DS       DS   
28        29                             Gran Turismo 3: A-Spec      PS2   
42        43                                       Mario Kart 7      3DS   
47        48                                     Gran Turismo 4      PS2   
...      ...                                                ...      ...   
16530  16533                           Sébastien Loeb Rally Evo     XOne   
16543  16546                             Driving Simulator 2011       PC   
16573  16576                                Mini Desktop Racing      Wii   
16574  16577  Yattaman Wii: BikkuriDokkiri Machine de Mou Ra...      Wii   
16595  16598   SCORE International Baja 1000: The Official Game      PS2   

       Year   Genre                    Publisher  NA_Sales  EU_Sales  \
2      2008  Racing                     Nintendo     15.85     12.88   
11     2005  Racing                     Nintendo      9.81      7.57   
28     2001  Racing  Sony Computer Entertainment      6.85      5.09   
42     2011  Racing                     Nintendo      4.74      3.91   
47     2004  Racing  Sony Computer Entertainment      3.01      0.01   
...     ...     ...                          ...       ...       ...   
16530  2016  Racing              Milestone S.r.l      0.00      0.01   
16543  2011  Racing                            0      0.00      0.01   
16573  2007  Racing               Popcorn Arcade      0.01      0.00   
16574  2008  Racing                  Takara Tomy      0.00      0.00   
16595  2008  Racing                   Activision      0.00      0.00   

       JP_Sales  Other_Sales  Global_Sales  
2          3.79         3.31         35.82  
11         4.13         1.92         23.42  
28         1.87         1.16         14.98  
42         2.67         0.89         12.21  
47         1.10         7.53         11.66  
...         ...          ...           ...  
16530      0.00         0.00          0.01  
16543      0.00         0.00          0.01  
16573      0.00         0.00          0.01  
16574      0.01         0.00          0.01  
16595      0.00         0.00          0.01  

[1249 rows x 11 columns]

In [286]:
race[race['Publisher']=='Sony Computer Entertainment']

Rank                                Name Platform  Year   Genre  \
28        29              Gran Turismo 3: A-Spec      PS2  2001  Racing   
47        48                      Gran Turismo 4      PS2  2004  Racing   
52        53                        Gran Turismo       PS  1997  Racing   
54        55                      Gran Turismo 5      PS3  2010  Racing   
69        70                      Gran Turismo 2       PS  1999  Racing   
...      ...                                 ...      ...   ...     ...   
12820  12822  MotoGP 4 - Official Game of MotoGP      PS2  2005  Racing   
13002  13004              Ferrari F355 Challenge      PS2  2002  Racing   
13893  13895   WRC: FIA World Rally Championship      PSP  2005  Racing   
13997  13999                        Formula 1 06      PS2  2006  Racing   
14166  14168                              MotoGP      PSP  2006  Racing   

                         Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
28     Sony Computer Entertainment      6.85      5.09      1.87         1.16   
47     Sony Computer Entertainment      3.01      0.01      1.10         7.53   
52     Sony Computer Entertainment      4.02      3.87      2.54         0.52   
54     Sony Computer Entertainment      2.96      4.88      0.81         2.12   
69     Sony Computer Entertainment      3.88      3.42      1.69         0.50   
...                            ...       ...       ...       ...          ...   
12820  Sony Computer Entertainment      0.03      0.02      0.00         0.01   
13002  Sony Computer Entertainment      0.03      0.02      0.00         0.01   
13893  Sony Computer Entertainment      0.02      0.01      0.00         0.01   
13997  Sony Computer Entertainment      0.00      0.00      0.04         0.00   
14166  Sony Computer Entertainment      0.03      0.00      0.00         0.00   

       Global_Sales  
28            14.98  
47            11.66  
52            10.95  
54            10.77  
69             9.49  
...             ...  
12820          0.06  
13002          0.05  
13893          0.04  
13997          0.04  
14166          0.03  

[65 rows x 11 columns]

In [33]:
games

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

### Q8 retrieve the average of global sales of nintendo publishers

In [39]:
pub=games.groupby(['Publisher'])

In [47]:
pubgl=pub.get_group('Nintendo')
pubgl

C:\Users\hp\AppData\Local\Temp\ipykernel_10700\4100238662.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  pubgl=pub.get_group('Nintendo')


Rank                                               Name Platform  \
0          1                                         Wii Sports      Wii   
1          2                                  Super Mario Bros.      NES   
2          3                                     Mario Kart Wii      Wii   
3          4                                  Wii Sports Resort      Wii   
4          5                           Pokemon Red/Pokemon Blue       GB   
...      ...                                                ...      ...   
16269  16272                           Slide Adventure: Mag Kid       DS   
16357  16360               Mario vs. Donkey Kong: Tipping Stars     WiiU   
16456  16459                           Art Academy: Home Studio     WiiU   
16473  16476                                    Captain Rainbow      Wii   
16542  16545  Mario & Luigi: Paper Jam & Mario Kart 7 Double...      3DS   

         Year         Genre Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports  Nintendo     41.49     29.02      3.77   
1      1985.0      Platform  Nintendo     29.08      3.58      6.81   
2      2008.0        Racing  Nintendo     15.85     12.88      3.79   
3      2009.0        Sports  Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing  Nintendo     11.27      8.89     10.22   
...       ...           ...       ...       ...       ...       ...   
16269  2007.0        Action  Nintendo      0.00      0.00      0.01   
16357  2015.0        Puzzle  Nintendo      0.00      0.00      0.01   
16456  2015.0          Misc  Nintendo      0.00      0.00      0.01   
16473  2008.0     Adventure  Nintendo      0.00      0.00      0.01   
16542  2015.0          Misc  Nintendo      0.00      0.00      0.01   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16269         0.00          0.01  
16357         0.00          0.01  
16456         0.00          0.01  
16473         0.00          0.01  
16542         0.00          0.01  

[703 rows x 11 columns]

In [53]:
pubgl['Global_Sales'].mean()

2.541337126600285

### Q9 retrieve the platform with highest globalsales

In [108]:
games.nlargest(1,'Global_Sales')

Rank        Name Platform    Year   Genre Publisher  NA_Sales  EU_Sales  \
0     1  Wii Sports      Wii  2006.0  Sports  Nintendo     41.49     29.02   

   JP_Sales  Other_Sales  Global_Sales  
0      3.77         8.46         82.74

### Q10 retrieve the details of the lowest globalsales in 2008

In [117]:
year=games.groupby(['Year'])

In [127]:
year_2008=year.get_group(2008)
year_2008

C:\Users\hp\AppData\Local\Temp\ipykernel_10700\728377944.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  year_2008=year.get_group(2008)


Rank                                              Name Platform  \
2          3                                    Mario Kart Wii      Wii   
39        40                           Super Smash Bros. Brawl      Wii   
51        52                               Grand Theft Auto IV     X360   
56        57                               Grand Theft Auto IV      PS3   
88        89                          Pokémon Platinum Version       DS   
...      ...                                               ...      ...   
16575  16578                             Neo Angelique Special      PSP   
16580  16583                                         Real Rode      PS2   
16588  16591                                  Mega Brain Boost       DS   
16592  16595                                          Plushees       DS   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   

         Year         Genre              Publisher  NA_Sales  EU_Sales  \
2      2008.0        Racing               Nintendo     15.85     12.88   
39     2008.0      Fighting               Nintendo      6.75      2.61   
51     2008.0        Action   Take-Two Interactive      6.76      3.10   
56     2008.0        Action   Take-Two Interactive      4.76      3.76   
88     2008.0  Role-Playing               Nintendo      2.82      1.78   
...       ...           ...                    ...       ...       ...   
16575  2008.0     Adventure             Tecmo Koei      0.00      0.00   
16580  2008.0     Adventure        Kadokawa Shoten      0.00      0.00   
16588  2008.0        Puzzle  Majesco Entertainment      0.01      0.00   
16592  2008.0    Simulation              Destineer      0.01      0.00   
16595  2008.0        Racing             Activision      0.00      0.00   

       JP_Sales  Other_Sales  Global_Sales  
2          3.79         3.31         35.82  
39         2.66         1.02         13.04  
51         0.14         1.03         11.02  
56         0.44         1.62         10.57  
88         2.69         0.55          7.84  
...         ...          ...           ...  
16575      0.01         0.00          0.01  
16580      0.01         0.00          0.01  
16588      0.00         0.00          0.01  
16592      0.00         0.00          0.01  
16595      0.00         0.00          0.01  

[1428 rows x 11 columns]

In [129]:
year_2008.nsmallest(1,"Global_Sales")

Rank   Name Platform    Year   Genre   Publisher  NA_Sales  EU_Sales  \
15980  15983  Turok       PC  2008.0  Action  Touchstone       0.0      0.01   

       JP_Sales  Other_Sales  Global_Sales  
15980       0.0          0.0          0.01

## Q11 retrieve the highest globalsales year which genre comes under action

In [132]:
genr=games.groupby(['Genre'])

In [138]:
gen=genr.get_group('Action')
gen

C:\Users\hp\AppData\Local\Temp\ipykernel_10700\2741153656.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  gen=genr.get_group('Action')


Rank                                               Name Platform  \
16        17                                 Grand Theft Auto V      PS3   
17        18                      Grand Theft Auto: San Andreas      PS2   
23        24                                 Grand Theft Auto V     X360   
24        25                        Grand Theft Auto: Vice City      PS2   
38        39                               Grand Theft Auto III      PS2   
...      ...                                                ...      ...   
16564  16567      Original Frisbee Disc Sports: Ultimate & Golf       DS   
16567  16570  Fujiko F. Fujio Characters: Great Assembly! Sl...      3DS   
16582  16585                                    Planet Monsters      GBA   
16583  16586                                     Carmageddon 64      N64   
16589  16592  Chou Ezaru wa Akai Hana: Koi wa Tsuki ni Shiru...      PSV   

         Year   Genre                  Publisher  NA_Sales  EU_Sales  \
16     2013.0  Action       Take-Two Interactive      7.01      9.27   
17     2004.0  Action       Take-Two Interactive      9.43      0.40   
23     2013.0  Action       Take-Two Interactive      9.63      5.31   
24     2002.0  Action       Take-Two Interactive      8.41      5.49   
38     2001.0  Action       Take-Two Interactive      6.99      4.51   
...       ...     ...                        ...       ...       ...   
16564  2007.0  Action  Destination Software, Inc      0.01      0.00   
16567  2014.0  Action         Namco Bandai Games      0.00      0.00   
16582  2001.0  Action                      Titus      0.01      0.00   
16583  1999.0  Action         Virgin Interactive      0.01      0.00   
16589  2016.0  Action            dramatic create      0.00      0.00   

       JP_Sales  Other_Sales  Global_Sales  
16         0.97         4.14         21.40  
17         0.41        10.57         20.81  
23         0.06         1.38         16.38  
24         0.47         1.78         16.15  
38         0.30         1.30         13.10  
...         ...          ...           ...  
16564      0.00         0.00          0.01  
16567      0.01         0.00          0.01  
16582      0.00         0.00          0.01  
16583      0.00         0.00          0.01  
16589      0.01         0.00          0.01  

[3316 rows x 11 columns]

In [144]:
gen.nlargest(1,'Global_Sales')

Rank                Name Platform    Year   Genre             Publisher  \
16    17  Grand Theft Auto V      PS3  2013.0  Action  Take-Two Interactive   

    NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
16      7.01      9.27      0.97         4.14          21.4

In [146]:
gen.nlargest(1,'Global_Sales')['Year']

16    2013.0
Name: Year, dtype: float64

### Q12 retrieve the games that comes under the platform 'PS3' 

In [150]:
plat=games.groupby(['Platform'])

In [156]:
plat.get_group('PS3')

C:\Users\hp\AppData\Local\Temp\ipykernel_10700\3514334872.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  plat.get_group('PS3')


Rank                                               Name Platform  \
16        17                                 Grand Theft Auto V      PS3   
34        35                         Call of Duty: Black Ops II      PS3   
37        38                     Call of Duty: Modern Warfare 3      PS3   
40        41                            Call of Duty: Black Ops      PS3   
54        55                                     Gran Turismo 5      PS3   
...      ...                                                ...      ...   
16450  16453                        Hyperdimension Neptunia mk2      PS3   
16451  16454  Shin Koihime Musou: Otome Taisen * Sangokushi ...      PS3   
16480  16483                                Muv-Luv Alternative      PS3   
16521  16524                           Akatsuki no Goei Trinity      PS3   
16541  16544                                    Battle Fantasia      PS3   

         Year       Genre                    Publisher  NA_Sales  EU_Sales  \
16     2013.0      Action         Take-Two Interactive      7.01      9.27   
34     2012.0     Shooter                   Activision      4.99      5.88   
37     2011.0     Shooter                   Activision      5.54      5.82   
40     2010.0     Shooter                   Activision      5.98      4.44   
54     2010.0      Racing  Sony Computer Entertainment      2.96      4.88   
...       ...         ...                          ...       ...       ...   
16450  2011.0      Action         Nippon Ichi Software      0.00      0.01   
16451  2014.0   Adventure                        Views      0.00      0.00   
16480  2012.0  Simulation                          5pb      0.00      0.00   
16521  2012.0   Adventure                          5pb      0.00      0.00   
16541  2008.0    Fighting                    505 Games      0.00      0.00   

       JP_Sales  Other_Sales  Global_Sales  
16         0.97         4.14         21.40  
34         0.65         2.52         14.03  
37         0.49         1.62         13.46  
40         0.48         1.83         12.73  
54         0.81         2.12         10.77  
...         ...          ...           ...  
16450      0.00         0.00          0.01  
16451      0.01         0.00          0.01  
16480      0.01         0.00          0.01  
16521      0.01         0.00          0.01  
16541      0.01         0.00          0.01  

[1329 rows x 11 columns]

In [158]:
plat.get_group('PS3')['Name']

C:\Users\hp\AppData\Local\Temp\ipykernel_10700\2329390583.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  plat.get_group('PS3')['Name']


16                                      Grand Theft Auto V
34                              Call of Duty: Black Ops II
37                          Call of Duty: Modern Warfare 3
40                                 Call of Duty: Black Ops
54                                          Gran Turismo 5
                               ...                        
16450                          Hyperdimension Neptunia mk2
16451    Shin Koihime Musou: Otome Taisen * Sangokushi ...
16480                                  Muv-Luv Alternative
16521                             Akatsuki no Goei Trinity
16541                                      Battle Fantasia
Name: Name, Length: 1329, dtype: object

## Q13 retrive the gamename with globalsales=20.81

In [169]:
games[games['Global_Sales']==20.81]

Rank                           Name Platform    Year   Genre  \
17    18  Grand Theft Auto: San Andreas      PS2  2004.0  Action   

               Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
17  Take-Two Interactive      9.43       0.4      0.41        10.57   

    Global_Sales  
17         20.81

In [171]:
games[games['Global_Sales']==20.81]['Name']

17    Grand Theft Auto: San Andreas
Name: Name, dtype: object

## Q14 list out the publishers in 2009

In [178]:
games[games['Year']==2009]

Rank                                  Name Platform    Year  \
3          4                     Wii Sports Resort      Wii  2009.0   
8          9             New Super Mario Bros. Wii      Wii  2009.0   
14        15                          Wii Fit Plus      Wii  2009.0   
36        37        Call of Duty: Modern Warfare 2     X360  2009.0   
45        46  Pokemon HeartGold/Pokemon SoulSilver       DS  2009.0   
...      ...                                   ...      ...     ...   
16540  16543                               15 Days       PC  2009.0   
16551  16554                    Sweet Honey Coming      PS2  2009.0   
16577  16580                             Damnation       PC  2009.0   
16586  16589        Secret Files 2: Puritas Cordis       DS  2009.0   
16587  16590              Mezase!! Tsuri Master DS       DS  2009.0   

           Genre          Publisher  NA_Sales  EU_Sales  JP_Sales  \
3         Sports           Nintendo     15.75     11.01      3.28   
8       Platform           Nintendo     14.59      7.06      4.70   
14        Sports           Nintendo      9.09      8.59      2.53   
36       Shooter         Activision      8.52      3.63      0.08   
45        Action           Nintendo      4.40      2.77      3.96   
...          ...                ...       ...       ...       ...   
16540  Adventure  DTP Entertainment      0.00      0.01      0.00   
16551  Adventure    Kadokawa Shoten      0.00      0.00      0.01   
16577    Shooter        Codemasters      0.00      0.01      0.00   
16586  Adventure        Deep Silver      0.00      0.01      0.00   
16587     Sports        Hudson Soft      0.00      0.00      0.01   

       Other_Sales  Global_Sales  
3             2.96         33.00  
8             2.26         28.62  
14            1.79         22.00  
36            1.29         13.51  
45            0.77         11.90  
...            ...           ...  
16540         0.00          0.01  
16551         0.00          0.01  
16577         0.00          0.01  
16586         0.00          0.01  
16587         0.00          0.01  

[1431 rows x 11 columns]

In [180]:
games[games['Year']==2009]['Publisher']

3                 Nintendo
8                 Nintendo
14                Nintendo
36              Activision
45                Nintendo
               ...        
16540    DTP Entertainment
16551      Kadokawa Shoten
16577          Codemasters
16586          Deep Silver
16587          Hudson Soft
Name: Publisher, Length: 1431, dtype: object

## Q15 list out the highest ranked publisher at 2006

In [189]:
games[games['Year']==2006].head(1)

Rank        Name Platform    Year   Genre Publisher  NA_Sales  EU_Sales  \
0     1  Wii Sports      Wii  2006.0  Sports  Nintendo     41.49     29.02   

   JP_Sales  Other_Sales  Global_Sales  
0      3.77         8.46         82.74

In [193]:
games[games['Year']==2006].head(1)['Publisher']

0    Nintendo
Name: Publisher, dtype: object